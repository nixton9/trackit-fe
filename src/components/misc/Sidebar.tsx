import React from 'react'
import { MenuIcon } from './MenuIcon'

const Sidebar: React.FC = () => {
  return (
    <>
      <p>Heyy</p>
      <MenuIcon />
    </>
  )
}

export default Sidebar
