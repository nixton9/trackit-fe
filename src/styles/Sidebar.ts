import styled from 'styled-components'

type SidebarContainerProps = {
  open: Boolean
}

const SidebarToggle = styled.div`
  cursor: pointer;
  position: absolute;
  top: 0.5rem;
  left: 1.5rem;
  fill: ${props => props.theme.white};

  svg {
    width: 2rem;
  }
`

const SidebarContainer = styled.div<SidebarContainerProps>`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  width: 65vw;
  max-width: 24rem;
  padding: 3rem 2rem;
  background: ${props => props.theme.mainGradient};
  box-shadow: 23px 0 16px 0 rgba(0, 0, 0, 0.07);
  transform: ${props => (props.open ? 'translateX(0);' : 'translateX(-101%);')};
  will-change: transform;
  transition: transform 0.2s linear;
  z-index: 11;
  overflow: hidden;
  svg {
    fill: ${props => props.theme.mainBlue};
    .svg-fill {
      fill: ${props => props.theme.mainBlue};
    }
    .svg-stroke {
      stroke: ${props => props.theme.mainBlue};
    }
  }
`

const SidebarOverlay = styled.div<SidebarContainerProps>`
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  overflow: hidden;
  display: ${props => (props.open ? 'block' : 'none')};
  z-index: 1;
`

const SidebarUser = styled.div`
  display: flex;
  align-items: center;
`

const SidebarUser__Img = styled.img`
  border-radius: 50%;
  width: 2.2rem;
  object-fit: contain;
`

const SidebarUser__Info = styled.div`
  margin-left: ${props => props.theme.spacingXXS};
`

const SidebarUser__Info__Name = styled.h3`
  color: ${props => props.theme.white};
  font-weight: ${props => props.theme.fontRegular};
  font-size: 0.9rem;
`

const SidebarUser__Info__Email = styled.p`
  color: ${props => props.theme.greyishBlue};
  font-weight: ${props => props.theme.fontSemiBold};
  font-size: 0.6rem;
`

const SidebarNav = styled.nav`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: ${props => props.theme.spacingS};
`

const SidebarNavItem = styled.div`
  display: flex;
  align-items: center;
  color: ${props => props.theme.white};
  margin-bottom: ${props => props.theme.spacingS};
  font-size: 1rem;
  font-weight: ${props => props.theme.fontRegular};

  &.settings {
    margin-bottom: 0;
  }

  svg {
    width: 1.9rem;
    margin-right: ${props => props.theme.spacingXXS};
    fill: ${props => props.theme.mainBlue};
    .svg-fill {
      fill: ${props => props.theme.mainBlue};
    }
    .svg-stroke {
      stroke: ${props => props.theme.mainBlue};
    }
  }
`

export const Styled = {
  SidebarToggle,
  SidebarContainer,
  SidebarOverlay,
  SidebarUser,
  SidebarUser__Img,
  SidebarUser__Info,
  SidebarUser__Info__Name,
  SidebarUser__Info__Email,
  SidebarNav,
  SidebarNavItem
}
