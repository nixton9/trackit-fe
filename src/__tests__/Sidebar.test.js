import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import Sidebar from '../components/misc/Sidebar'

let container = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

it('matches snapshot', () => {
  act(() => {
    render(<Sidebar />, container)
  })

  expect(container.innerHTML).toMatchSnapshot()
})
