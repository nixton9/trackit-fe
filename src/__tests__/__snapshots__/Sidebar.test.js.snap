// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `"<p>Heyy</p><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" width=\\"69\\" height=\\"58\\" viewBox=\\"0 0 69 58\\"><defs><clipPath id=\\"clip-Prancheta_4\\"><rect width=\\"69\\" height=\\"58\\"></rect></clipPath></defs><g id=\\"Prancheta_4\\" data-name=\\"Prancheta – 4\\" clip-path=\\"url(#clip-Prancheta_4)\\"><g id=\\"Grupo_150\\" data-name=\\"Grupo 150\\" transform=\\"translate(-38 -63)\\"><rect id=\\"Retângulo_139\\" data-name=\\"Retângulo 139\\" width=\\"63\\" height=\\"7\\" rx=\\"3.5\\" transform=\\"translate(41 68)\\"></rect><rect id=\\"Retângulo_140\\" data-name=\\"Retângulo 140\\" width=\\"36\\" height=\\"7\\" rx=\\"3.5\\" transform=\\"translate(41 89)\\"></rect><rect id=\\"Retângulo_141\\" data-name=\\"Retângulo 141\\" width=\\"52\\" height=\\"7\\" rx=\\"3.5\\" transform=\\"translate(41 110)\\"></rect></g></g></svg>"`;
